src/routes/auth/tokens.js

```diff
@@ -45,20 +48,29 @@ module.exports = async (server, options) => {
             notes: `This endpoint will create a new API Token and show it in the response body.
                      It is possible to create a comment with every token to have a reference where it is used.
                      Make sure to save the token somewhere, since you won't be able to see it again.`,
+            auth: {
+                scope: ['auth', 'all']
+            },
             validate: {
                 payload: Joi.object({
                     comment: Joi.string()
                         .required()
                         .example('Token for fun project')
                         .description(
                             'The comment can be everything. Tip: Use something to remember where this specific token is used.'
-                        )
+                        ),
+                    scopes: Joi.array().items(
+                        Joi.string()
+                            .regex(/^[a-z-]+$/)
+                            .description('scopes to be granted for this token')
+                    )
```

Comment:
You could use `server.app.scopes` here to get all defined scopes.

```suggestion
                    scopes: Joi.array().items(
                        Joi.string()
                             .valid(...server.app.scopes)
                            .description('scopes to be granted for this token')
                    )
```

Reply:
nope that doesn't work. tried it before but the problem is that by the time the `auth/token` routes are initialized not all other route handlers have been executed. that means that both scopes defined in the api as well as scopes defined in plugins won't be included in the Joi.schema.

I then tried delaying the definition of that particular route, but I can't delay it until after the plugins are loaded because plugins already wait for the core routes.

Also this introduced problems with the swagger endpoint which is also defined at some point and my "delayed" routes weren't in there (neither are routes defined in plugins btw).

Reply:
we had the same problem with other dynamic validations, such as chart type. see here https://github.com/datawrapper/api/blob/api-token-scopes/src/routes/charts/index.js#L206-L211

Reply:
Ah I see. That's too bad.

Reply:
would be nice if there was a way to define a dynamic schema